certtools/keys.go

```diff
@@ -76,6 +99,40 @@ func GetKey(eng *url.URL, file, newkey string) (crypto.PrivateKey, error) {
 	return nil, fmt.Errorf("no key mechanism specified")
 }
 
+func verifyExistingKey(file string, existingKey crypto.PrivateKey, newkey string) (crypto.PrivateKey, error) {
+	//desired type specified, verify it
+	specs := strings.Split(newkey, ":")
```

Comment:
i feel like being defensive here is a good idea since you have specs[1] referenced below. checking for ":" or len of specs or something?

Reply:
Done.